### PR TITLE
Delete api token

### PR DESCRIPTION
**Pull request by FabianDevel**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/utils/AccountUtils.kt [hunk 1]
```diff
@@ -301,4 +315,8 @@ object AccountUtils : CredentialManager {
     }
 
     fun isEnableAppSync(): Boolean = UploadFile.getAppSyncSettings() != null
+
+    fun initInfomaniakLogin(context: Context): InfomaniakLogin {
+        return InfomaniakLogin(context, appUID = BuildConfig.APPLICATION_ID, clientID = BuildConfig.CLIENT_ID)
+    }
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/drive/utils/AccountUtils.kt:321]: It should be in a `companion object`.
And be called `getInfomaniakLogin`.
It's not really an init

And maybe even an extension of Context

File: app/src/main/java/com/infomaniak/drive/ui/login/LoginActivity.kt [hunk 2]
```diff
@@ -96,11 +95,12 @@ class LoginActivity : AppCompatActivity() {
...
                 }
             })
+
+            nextButton.setOnClickListener { currentItem++ }
         }
 
-        dotsIndicator.setViewPager2(introViewpager)
+        dotsIndicator.attachTo(introViewpager)
 
-        nextButton.setOnClickListener { introViewpager.currentItem = introViewpager.currentItem + 1 }
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/drive/ui/login/LoginActivity.kt:104]: Why did you do this?
And what does this have to do with RA?

File: app/src/main/java/com/infomaniak/drive/utils/AccountUtils.kt [hunk 3]
```diff
@@ -193,16 +195,28 @@ object AccountUtils : CredentialManager {
...
-        userDatabase.userDao().delete(userRemoved)
-        FileController.deleteUserDriveFiles(userRemoved.id)
+    suspend fun removeUserAndDeleteToken(context: Context, user: User) {
+        initInfomaniakLogin(context).deleteToken(
+            KDriveHttpClient.getHttpClient(user.id),
+            user.apiToken,
+            onError = {
+                Log.e("deleteTokenError", "Api response error : ${LoginActivity.getLoginErrorDescription(context, it)}")
+            },
+        )
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/drive/utils/AccountUtils.kt:205]: So if the request takes too long to answer, the person will be blocked in the app?
It has to be in another thread.

**Review by JorisBodin — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/ui/login/LoginActivity.kt (hunk 2 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/ui/login/LoginActivity.kt:104]: I did this because setViewPager2() is deprecated and must be replace with attachTo(), and  the listener used 2 times introViewPager to update one of its attributes and there is already an introViewPager.apply {} right above (ending line 100).

It could have been in another PR but I kept it in this one because it's in the same function that I was already modifying and it's a small change. I can deplace it in another PR if needed.

File: app/src/main/java/com/infomaniak/drive/utils/AccountUtils.kt [hunk 4]
```diff
@@ -193,16 +195,28 @@ object AccountUtils : CredentialManager {
...
-    suspend fun removeUser(context: Context, userRemoved: User) {
-        userDatabase.userDao().delete(userRemoved)
-        FileController.deleteUserDriveFiles(userRemoved.id)
+    suspend fun removeUserAndDeleteToken(context: Context, user: User) {
+        initInfomaniakLogin(context).deleteToken(
+            KDriveHttpClient.getHttpClient(user.id),
+            user.apiToken,
+            onError = {
+                Log.e("deleteTokenError", "Api response error : ${LoginActivity.getLoginErrorDescription(context, it)}")
+            },
```
[the remark this answers is not in the document]
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/utils/AccountUtils.kt]: done

File: app/src/main/java/com/infomaniak/drive/ui/login/LoginActivity.kt (hunk 2 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/ui/login/LoginActivity.kt]: done in #925

File: app/src/main/java/com/infomaniak/drive/utils/AccountUtils.kt (hunk 3 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/utils/AccountUtils.kt]: done

File: app/src/main/java/com/infomaniak/drive/utils/AccountUtils.kt (hunk 1 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/utils/AccountUtils.kt]: Done (moved in  Extension.kt)

**Review by JorisBodin — APPROVED** [review]
